packages/preferences/src/browser/abstract-resource-preference-provider.ts

```diff
@@ -113,6 +117,9 @@ export abstract class AbstractResourcePreferenceProvider extends PreferenceProvi
         if (!this.model) {
             return false;
         }
+        if (this.model.dirty) {
```

Comment:
Preference model can be dirty for other reasons, like previous `setPreference` request is not finished yet. In such cases it behaves correctly as on master and I afraid this PR will cause race conditions and bogus notifications. Do we care only about cases when there is an opened editor for dirty model?

Reply:
Yes, the problem that this PR aims to solve only occurs when an editor is open, as `applyBackgroundEdit` will not save changes in that case. The basic cases are:

1. Editor not open -> make changes & save
1. Editor open, not dirty -> make changes & save
1. Editor open and dirty -> make no changes, alert user

Is there a better / more reliable way to make the checks necessary to achieve that behavior?

Reply:
the current logic could be fine as well, the issue is that we cannot accept the PR without checking that it does not introduce regressions under other conditions.

We have api-samples package, you can add a command where which trigger many concurrent updates to preferences and another command to stop it. We can use such command to test under different conditions manually.

Or we have api-tests package, where you can add integration tests to verify different cases.

Reply:
👍 , I will have a look at the API-Tests and see how I can simulate a stress test.